plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/FeatureDiagramEditor.java

```diff
@@ -655,88 +655,83 @@ private void fillContextMenu(IMenuManager menu) {
+
+		// if one or more features are selected
+		else if ((createLayerAction.isEnabled()	|| createCompoundAction.isEnabled()) && !connectionSelected) {
 			menu.add(createCompoundAction);
 			menu.add(createLayerAction);
 			menu.add(createConstraintWithAction);
 			menu.add(renameAction);
+			menu.add(changeFeatureDescriptionAction);
 			menu.add(deleteAction);
 			menu.add(deleteAllAction);
 			menu.add(new Separator());
 			connectionEntrys(menu);
 			menu.add(mandatoryAction);
 			menu.add(abstractAction);
 			menu.add(hiddenAction);
+			menu.add(colorSelectedFeatureAction);
+			menu.add(new Separator());
 			menu.add(collapseAction);
 			menu.add(collapseFeaturesAction);
 			if (getActiveExplanation() != null) {
 				menu.add(collapseAllButExplanationAction);
 			}
-			menu.add(changeFeatureDescriptionAction);
-			menu.add(new Separator());
-			menu.add(subMenuLayout);
-			menu.add(subMenuCalculations);
+
 			menu.add(new Separator());
 			menu.add(calculateDependencyAction);
-			menu.add(reverseOrderAction);
-			menu.add(legendAction);
 			menu.add(new Separator());
-		} else if (editConstraintAction.isEnabled() && !connectionSelected) {
+
+      // if a constraint is selected
+		} else if (editConstraintAction.isEnabled()	&& !connectionSelected) {
+
 			menu.add(createConstraintAction);
-			menu.add(expandConstraintAction);
 			menu.add(editConstraintAction);
 			menu.add(deleteAction);
+			menu.add(new Separator());
+			menu.add(expandConstraintAction);
 			if (getActiveExplanation() != null) {
 				menu.add(new Separator());
 				menu.add(collapseAllButExplanationAction);
 			}
+
+			// if the legend is selected
 		} else if (legendLayoutAction.isEnabled()) {
 			menu.add(legendLayoutAction);
 			menu.add(legendAction);
+
+			// if a connection is selected
 		} else if (andAction.isEnabled() || orAction.isEnabled() || alternativeAction.isEnabled()) {
+
 			connectionEntrys(menu);
+
+			// if nothing is selected
 		} else {
 			menu.add(createConstraintAction);
 			menu.add(new Separator());
 			menu.add(collapseAllAction);
 			menu.add(expandAllAction);
+			menu.add(showHiddenFeaturesAction);
+			menu.add(showCollapsedConstraintsAction);
 			menu.add(adjustModelToEditorSizeAction);
 			menu.add(new Separator());
 			menu.add(subMenuLayout);
 			menu.add(subMenuCalculations);
 			menu.add(new Separator());
 			menu.add(reverseOrderAction);
 			menu.add(legendAction);
-		}
 
-		boolean isEmpty = true;
-		for (final Object obj : ((StructuredSelection) getSelection()).toArray()) {
-			if ((obj instanceof FeatureEditPart) || (obj instanceof IFeature)) {
-				isEmpty = false;
-			}
-		}
-		if (!isEmpty) {
-			menu.add(new Separator());
-			menu.add(colorSelectedFeatureAction);
-		}
-
-		menu.add(new Separator(IWorkbenchActionConstants.MB_ADDITIONS));
-		if (featureModelEditor.getFeatureModel().getStructure().hasHidden()) {
+			// if there are hidden features and nothing is selected
```

Comment:
redundant to "// if nothing is selected" above this else block
besides that, this comment seems to be misleading because that export menu item is displayed even if there are no hidden features

Reply:
Right. We forgot to remove it after restructuring.